code/modules/hydroponics/gene_modder.dm

```diff
@@ -492,15 +492,25 @@
 	. = ..()
 	if(HAS_TRAIT(src, TRAIT_CMAGGED))
 		desc = "Better keep this safe."
-	else
-		desc = "A disk for storing plant genetic data."
+		return
+
+	desc = "A disk for storing plant genetic data."
 
 /obj/item/disk/plantgene/update_icon_state()
 	. = ..()
 	if(HAS_TRAIT(src, TRAIT_CMAGGED))
 		icon_state = "nucleardisk"
-	else
-		icon_state = "datadisk_hydro"
+		return
+
+	icon_state = "datadisk_hydro"
+
+/obj/item/disk/plantgene/update_overlays()
+	. = ..()
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))
+		overlays -= "datadisk_gene"
+		return
+
+	overlays += "datadisk_gene"
```

Comment:
This proc should add the overlays to the return value, instead of `overlays` directly.
```suggestion
		. -= "datadisk_gene"
		return

	. += "datadisk_gene"
```

Reply:
Replacing `overlays` with the return value breaks the proc; cmagged disks do not lose or regain their overlay when cmag status is added or removed. Any clue why this might be the case?

Reply:
Fixed, see below